src/utils/cspUtils.js

```diff
@@ -0,0 +1,133 @@
+// Supported CSP checks are img-src, frame-src, media-src, object-src script-src-elem 
+// Supported fallbacks are child-src, script-src, and default-src
+
+import toml from 'toml';
+import axios from 'axios';
+import Policy from 'csp-parse';
+import _ from 'lodash';
+import { isLinkInternal } from '../utils';
+import cheerio from 'cheerio';
+import escapeStringRegexp from 'escape-string-regexp';
+
+/* Helper function to retrieve the netlify.toml from repo */
+async function _parseNetlifyToml(repoName) {
+  // axios get withCredentials false is required https://stackoverflow.com/questions/34078676/access-control-allow-origin-not-allowed-when-credentials-flag-is-true-but/34099399 
+  const tomlUrl = `https://raw.githubusercontent.com/isomerpages/${repoName}/staging/netlify.toml`;
+  const tomlFile = await axios.get(tomlUrl, { withCredentials: false } ); 
+  return toml.parse(tomlFile.data);
+};
+
+export async function getCSP(repoName) {
+  const tomlData = await _parseNetlifyToml(repoName);
+  const csp = tomlData.headers[0].values['Content-Security-Policy'];
+  return csp;
+};
+
+function _stringContainsValue(string, value) {
+  const VALUE_REGEX = `(\\s|^)${value}(\\s|$)`;
+  const ValueRegexTest = new RegExp(VALUE_REGEX);
+  return ValueRegexTest.test(string);
+};
+
+function _toRegExp(string) {
+  const strippedString = string.replace(/\/$/, '') // removes ending '/' from domains eg 'abc.com/'
+  const escapedStrippedString = escapeStringRegexp(strippedString) // add escape characters 
+  const wildcardEscapedStrippedString = escapedStrippedString.replace('\\*', '[a-z0-9-]*') // convert wildcards
+  const RegExpString = RegExp(wildcardEscapedStrippedString) // converts to RegExp
+  return RegExpString;
+};
+
+/* Helper functions to check if elemSrc satisfies each CSP source specification: host-source, schema-source and 'self' */
+function _checkHostsourcePolicy (elemSrc, policy) {
+  if (_stringContainsValue(policy, '\\*')) return true;
+
+  const specialValues = ['http:', 'https:', 'data:', 'mediastream:', 'blob:', 'filesystem:', "'self'", "'none'", '*'];
+  const hostsources = policy.split(' ').filter(value => (!specialValues.includes(value)));
+  
+  const hostsourcesSatisfied = hostsources.some(hostsource => (_toRegExp(hostsource).test(elemSrc)));
+  return hostsourcesSatisfied;
+};
+
+function _checkSchemasourcePolicy (elemSrc, policy) {
+  const schemes = ['http:', 'https:'];
+  const dataschemes = ['data:', 'mediastream:', 'blob:', 'filesystem:'];
```

Comment:
nit: convert `dataschemes` variable name to camelCase

Reply:
e0b1d3c